### PR TITLE
store inputs

### LINKED ISSUE #4: Storing input boxes and specifying spent boxes

### PR DESCRIPTION
**Pull request by code-for-uss**
Related to #4 
Some Changes:
  - Added functionality for storing input boxes  related to output boxes and specifying spent boxes
  - Fixed some todos about DB blocking
  - Add `dataInputs` DAO and functionality for store them
  - Fixed bug duplicate scanning output boxes
  - Merged all migrations into one schema

### CODE REVIEW

File: app/dao/ExtractionResultDAO.scala [hunk 1]
```diff
@@ -1,46 +1,84 @@
...
+import scala.concurrent.{Await, ExecutionContext}
 
 
-class ExtractionResultDAO @Inject() (extractedBlockDAO: ExtractedBlockDAO, transactionDAO: TransactionDAO, inputDAO: InputDAO, outputDAO: OutputDAO, assetDAO: AssetDAO, registerDAO: RegisterDAO, protected val dbConfigProvider: DatabaseConfigProvider) (implicit executionContext: ExecutionContext)
+class ExtractionResultDAO @Inject() (extractedBlockDAO: ExtractedBlockDAO, transactionDAO: TransactionDAO, dataInputDAO: DataInputDAO, inputDAO: InputDAO, outputDAO: OutputDAO, assetDAO: AssetDAO, registerDAO: RegisterDAO, protected val dbConfigProvider: DatabaseConfigProvider) (implicit executionContext: ExecutionContext)
     extends DbUtils with HasDatabaseConfigProvider[JdbcProfile] {
 
+  import profile.api._
   /**
   * store data into db as transactionally.
```
Comment (kushti) [inline app/dao/ExtractionResultDAO.scala]: Unfinished description

File: app/dao/ExtractionResultDAO.scala [hunk 2]
```diff
@@ -1,46 +1,84 @@
...
-   * @param extractionResult : ExtractionResultModel extracted data
+   * @param spentTrackedInputs : ExtractionInputResultModel extracted spent Tracked Inputs
+   */
+  def storeBaseOnInputs(spentTrackedInputs: Seq[ExtractionInputResultModel]): Unit = {
```
Comment (kushti) [inline app/dao/ExtractionResultDAO.scala]: Sounds like inputs for tracked outputs being spent passed, while in fact all the inputs in the block. Please fix descriptions and naming. 

Inputs filtering can be very inefficient, can we consider using and storing a Bloom filter maybe?

**Review by kushti — CHANGES_REQUESTED** [review]

File: app/dao/ExtractionResultDAO.scala (hunk 2 above)
Comment (code-for-uss) [reply app/dao/ExtractionResultDAO.scala]: I think according to the database queries, this is the only way to use Bloom filter, If you have another comment please share

File: app/dao/ExtractionResultDAO.scala (hunk 1 above)
Comment (code-for-uss) [reply app/dao/ExtractionResultDAO.scala]: Change naming and descriptions

File: app/dao/ExtractionResultDAO.scala (hunk 2 above)
Comment (kushti) [reply app/dao/ExtractionResultDAO.scala]: It seems Bloom filter is not doing doing much here, also, after checking with mightContain you need to check for false positives. Let's revert ?

Comment (code-for-uss) [reply app/dao/ExtractionResultDAO.scala]: Revert on which? the filter or check false positives for Bloom filter? 
I think revert on the filter is better!

**Review by kushti — APPROVED** [review]

**Pull request merged**